gateway-api: add watch for reference grant in TLSRoute reconciler

mhofstetter opened the pull request:
This commit adds the missing reference-grant watch to the TLS route controller.

Fixes: #25573 & #25711

mhofstetter committed:
gateway-api: tls route: add watch for reference grant

This commit adds the missing reference-grant watch to the TLS route controller.

Fixes: #25573 & #25711

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

meyskens commented on operator/pkg/gateway-api/tlsroute.go:
```diff
@@ -102,6 +102,11 @@ func (r *tlsRouteReconciler) enqueueRequestForBackendService() handler.EventHand
 	return handler.EnqueueRequestsFromMapFunc(r.enqueueFromIndex(backendServiceIndex))
 }
 
+// enqueueRequestForRequestGrant makes sure that TLS Routes in the same namespace are reconciled
```
This comment feels outdated?

mhofstetter replied:
regarding the part about `in the same namespace` ? -> `all TLS Routes are reconciled`

edit: + `RequestGrant` -> `RefererenceGrant`

mhofstetter replied:
renamed function and adapted comment in all route reconcilers

meyskens reviewed (approved):
Small nit, LGTM overall

mhofstetter committed:
gateway-api: rename enqueueRequestForReferenceGrant

This commit fixes the name of the function `enqueueRequestReferenceGrant`
in all the route reconcilers.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>
